File: java/demo/src/main/java/com/yuanzhibang/demo/RedisHelper.java
package com.yuanzhibang.demo;

import java.util.Map;

import com.google.gson.Gson;
import com.lambdaworks.redis.*;

public class RedisHelper {
    public String redisUrl;

    RedisHelper(String redisUrl) {
        this.redisUrl = redisUrl;
    }

    public String getJsTicket(String appId) {
        String key = RedisHelper.getJsTicketCacheKey(appId);
        return this.getRedisJsonValueByKey(key, "js_ticket");
    }

    public String getServerAccessToken(String appId) {
        String key = RedisHelper.getServerAccessTokenCacheKey(appId);
        return this.getRedisJsonValueByKey(key, "server_access_token");
    }

    public static String getServerAccessTokenCacheKey(String appId) {
        return String.format("type/server_access_token_info/app_id/%s", appId);
    }

    public static String getJsTicketCacheKey(String appId) {
        return String.format("type/js_ticket_info/app_id/%s", appId);
    }

    @SuppressWarnings("unchecked")
    public String getRedisJsonValueByKey(String key, String codeKey) {
        RedisClient redisClient = new RedisClient(
                RedisURI.create(this.redisUrl));
        RedisConnection<String, String> connection = redisClient.connect();
        String jsonInfo = connection.get(key);
        Gson gson = new Gson();
        Map<String, String> info = gson.fromJson(jsonInfo, Map.class);
        String value = info.get(codeKey);
        connection.close();
        redisClient.shutdown();
        return value;
    }
}

File: php/RedisHelper.php
<?php

class RedisHelper
{
    public $config;
    public function __construct($config)
    {
        $this->config = $config;
    }

    public function getJsTicket($appId)
    {
        $key = RedisHelper::getJsTicketCacheKey($appId);
        $value = $this->getRedisJsonValueByKey($key, 'js_ticket');
        return $value;
    }

    public function getServerAccessToken($appId)
    {
        // 请通过apt install php7.4-redis 类似的语句或者自己编译的方式来安装php的redis拓展
        $key = RedisHelper::getServerAccessTokenCacheKey($appId);
        $value = $this->getRedisJsonValueByKey($key, 'server_access_token');
        return $value;
    }

    static public function getServerAccessTokenCacheKey($appId)
    {
        return "type/server_access_token_info/app_id/$appId";
    }
    static public function getJsTicketCacheKey($appId)
    {
        return "type/js_ticket_info/app_id/$appId";
    }
    public function getRedisJsonValueByKey($key, $codeKey)
    {
        // 请通过apt install php7.4-redis 类似的语句或者自己编译的方式来安装php的redis拓展,具体的redis用法要参照不同的phpredis 版本
        $redis = new Redis();
        $redis->connect($this->config['host'], $this->config['port'], 10);
        $redis->auth($this->config['auth']);
        $redis->select(0);
        $infoString = $redis->get($key);
        $info = json_decode($infoString, true);
        $result = $info[$codeKey];
        $redis->close();
        return $result;
    }
}


$config = [
    "host" => "demo-dev-cache-redis",
    "port" => 6379,
    "auth" => 'p8WOmXgzZg',
];
$redisHelper = new RedisHelper($config);
$jsTicket = $redisHelper->getJsTicket("100027");
$serverAccessToken = $redisHelper->getServerAccessToken("100027");
var_dump($jsTicket);
var_dump($serverAccessToken);
